metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py

```diff
@@ -383,6 +386,32 @@ def from_exported_bigquery_audit_metadata(
         return query_event
 
 
+class BigQueryCredential(ConfigModel):
+    project_id: str
+    private_key_id: str
+    private_key: str
+    client_email: str
+    client_id: str
+    auth_uri: str = "https://accounts.google.com/o/oauth2/auth"
+    token_uri: str = "https://oauth2.googleapis.com/token"
+    auth_provider_x509_cert_url: str = "https://www.googleapis.com/oauth2/v1/certs"
+    type: str = "service_account"
+    client_x509_cert_url: Optional[str]
+
+    def __init__(self, **data: Any):
```

Comment:
Could you remove he `init` method, and use a pydantic root validator to do this, similar to [this](https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/state/stateful_ingestion_base.py#L50).

Reply:
I updated the pr but I had to stick to __init__ because I changed `credentials_path` to private attribute and that I could not set in root validator.